lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -278,6 +280,25 @@ class Module < Namespace
       end
 
       class Class < Namespace
+        extend T::Sig
+
+        # The unresolved name of the parent class
+        sig { returns(String) }
+        attr_reader :parent_class
+
+        sig do
+          params(
+            name: String,
+            file_path: String,
+            location: YARP::Location,
+            comments: T::Array[String],
+            parent_class: T.nilable(String),
+          ).void
+        end
+        def initialize(name, file_path, location, comments, parent_class)
+          super(name, file_path, location, comments)
+          @parent_class = T.let(parent_class || "Object", String)
```

Comment:
That's a good point. We should store `nil` instead since you can have this

```ruby
module Foo
  module Bar
    class Baz < Object
    end
  end
end
```

In this scenario, the parent class might be `Foo::Bar::Object`, `Foo::Object` or `Object`. But if the super class is omitted, then it's 100% sure `Object` even if the other two object classes exist.

I changed to store `nil` instead.